src/config/vars.rs

```diff
@@ -16,27 +21,63 @@ pub fn interpolate(input: &str, vars: &HashMap<String, String>) -> (String, Vec<
         r"(?x)
         \$\$|
         \$([[:word:].]+)|
-        \$\{([[:word:].]+)(?::-([^}]+)?)?\}",
+        \$\{([[:word:].]+)(?:(:?-|:?\?)([^}]*))?\}",
     )
     .unwrap();
+
     let interpolated = re
         .replace_all(input, |caps: &Captures<'_>| {
+            let flags = caps.get(3).map(|m| m.as_str()).unwrap_or("");
+            let def_or_err = caps.get(4).map(|m| m.as_str()).unwrap_or("");
```

Comment:
```suggestion
            let flags = caps.get(3).map(|m| m.as_str()).unwrap_or_default();
            let def_or_err = caps.get(4).map(|m| m.as_str()).unwrap_or_default();
```

Reply:
Ah, doh, indeed `unwrap_or_default()` will resolve to `""` and is cleaner. Thanks!